### PR TITLE
APEXMALHAR-2366 #resolve #comment Apply BloomFilter to Bucket, use internal BloomFilter

### PR DESCRIPTION
**Pull request by pramodin**
@bhupeshchawda please see, this is to finish up the work started in https://github.com/apache/apex-malhar/pull/521

### CODE REVIEW

**Review by bhupeshchawda — APPROVED** [review]

**Comment by bhupeshchawda** [comment]
Will merge this today.

**Comment by bhupeshchawda** [comment]
Can you please add the JIRA id to the second commit message?

File: library/src/test/java/org/apache/apex/malhar/lib/state/managed/SliceBloomFilterTest.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+      public void populateDAG(DAG dag, Configuration conf)
+      {
+      }
+    };
+
+    lma.prepareDAG(app, conf);
+
+    // Create local cluster
+    final LocalMode.Controller lc = lma.getController();
+    lc.run(3000);
```
Comment (tweise) [inline library/src/test/java/org/apache/apex/malhar/lib/state/managed/SliceBloomFilterTest.java:187]: @bhupeshchawda can you please look at this test and remove the timing dependency?

Comment (bhupeshchawda) [reply library/src/test/java/org/apache/apex/malhar/lib/state/managed/SliceBloomFilterTest.java:187]: This test was supposed to just test serialization, not a timing dependency as such. 
But perhaps the test is not sufficient. Will look into it.

**Pull request merged**